### PR TITLE
Upgrade @jupyterlab/services@0.50.0

### PR DESCRIPTION
**Pull request by lgeiger**
Thanks @blink1073 for the sessions backport 👍 

Closes #1042
Closes #781
Closes #908

### CODE REVIEW

File: lib/ws-kernel.js [hunk 1]
```diff
@@ -122,7 +122,7 @@ export default class WSKernel extends Kernel {
         defaultText: this.session.path,
         allowCancel: true
       },
-      (input: string) => this.session.rename(input)
+      (input: string) => this.session.setPath(input)
```
Comment (lgeiger) [inline lib/ws-kernel.js]: This doesn't change behavior. Though we should now use the session name in the UI instead of the notebook path.

Comment (blink1073) [reply lib/ws-kernel.js]: We use the name unless it is an empty string and then defer to the `basename` of the path.

**Comment by blink1073** [comment]
Looks good!  You can update to `0.50.1` which adds source maps to the released package that were supposed to be there in `0.50.0`.

**Comment by blink1073** [comment]
Missed some of the source maps in 0.50.1, 0.50.2 has them all.

**Comment by lgeiger** [comment]
Thanks, lets get up to date 🎉

**Pull request merged**